src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkCustomCommandLine.java

```diff
@@ -1377,10 +1382,12 @@ public void repr(Printer printer) {
   }
 
   /** An element in a {@link PreprocessedCommandLine}. */
-  private interface PreprocessedArg extends Iterable<String> {
+  private interface PreprocessedArg {
+    Iterator<String> iterator(PathMapper pathMapper);
```

Comment:
I think it would be cleaner to change this to `Iterable<String> toIterable(PathMapper pathMapper)`. That would get rid of some awkward nested lambdas. Most of the implementations can return a singleton `ImmutableList`.